magma/conversions.py

```diff
@@ -35,6 +35,36 @@
 __all__ += ['replace']
 __all__ += ['as_bits', 'from_bits']
 
+
+class _Concatter(Finalizable):
+    def __init__(self):
+        self._ts = []
+        self._types = []
+
+    def _consume_impl(self, value):
+        if isinstance(value, ht.BitVector):
+            return value.bits(), Bits
+        if isinstance(magma_value(value), Bit):
```

Comment:
Should we just call `magma_value` once at the top?

Reply:
I'm not sure if that works with the checking for BitVector, bool cases here. Also, this maintains parity with the old code. I agree this should be made more uniform (not really clear where protocol types are allowed here). Will keep for now.